lib/CoreBot.js

```diff
@@ -568,10 +576,13 @@ function Botkit(configuration) {
                         (duration > this.task.timeLimit) && // timelimit is up
                         (lastActive > this.task.timeLimit) // nobody has typed for 60 seconds at least
                     ) {
-
-                        if (this.threads.timeout) {
+                        // if timeoutHandler is set then call it, otherwise follow the normal flow
+                        // this will not break others code, after the update
+                        if (this.timeOutHandler != null || this.timeOutHandler != undefined) {
```

Comment:
A shorter version could be `if (this.timeOutHandler)` since we're only checking for non null and defined conditions.